Rework VTK rendering integration & simplify UI

Ozaq opened the pull request:
There are still several rendering issues and rance conditions lurking,
but right now this should already improve the situation quite a bit.

schroedtert commented on src/InteractorStyle.h:
```diff
@@ -28,9 +28,8 @@
-
-#ifndef INTERACTORSTYLE_H_
-#define INTERACTORSTYLE_H_
+#pragma once
```
Will this be changed in every file with the upcoming housekeeping PR?

Ozaq replied:
Yes I would like to do this for all files at some point.

schroedtert commented on src/main.cpp:
```diff
@@ -93,42 +89,34 @@ std::string true_cxx_ver =
+    setlocale(LC_NUMERIC, "C");
+
+    // force the application to first looks for privated libs
+    a.addLibraryPath(QApplication::applicationDirPath() + QDir::separator() +
+                     "lib");
```
Do we need this?

Ozaq replied:
Not sure if this is needed to find dependencies in installed packages, hence I left it until I had time to check further.

schroedtert commented on src/ThreadVisualisation.h:
```diff
@@ -76,16 +78,16 @@ extern vtkActor* extern_glyphs_pedestrians_actor_3D;
 
 
-class ThreadVisualisation :public QThread {
+class ThreadVisualisation :public QObject {
```
Why the change from `QThread` to `QObject`? 
Maybe also a renaming of `ThreadVisualisation` should be considered, it is not a thread anymore, right?

Ozaq replied:
The change from `QThread` to `QObject` was done because this is no longer spawning a thread but still exposing signals and slots. I would like to distribute the contents of the class into different ones. But I think your right, even if it is just for some time, the class should be renamed. Ill rename it to `Visualisation` not good but at least it does no longer imply it would be a thread.

schroedtert commented on src/MainWindow.cpp:
```diff
@@ -227,789 +180,603 @@ MainWindow::MainWindow(QWidget *parent) :
-            if(argument.compare("help")==0) {
-                Log::Info("Usage: jpsvis [file1] [-2D] [-caption] [-online [port]]");
+            if (argument.compare("help") == 0) {
+                Log::Info(
+                    "Usage: jpsvis [file1] [-2D] [-caption] [-online [port]]");
```
Is `-online [port]` still supported?

Ozaq replied:
nope, will remove it

chraibi commented:
I think  VTK8's `GetRenderWindow()` was renamed in VTK9 to `renderWindow()`. See [diff](https://fossies.org/diffs/VTK/8.2.0_vs_9.0.0/GUISupport/Qt/QVTKOpenGLNativeWidget.h-diff.html)
and CI @8.2 is choking 
<img width="1118" alt="Screen Shot 2021-09-26 at 06 04 19" src="https://user-images.githubusercontent.com/5772973/134793302-833f166d-be72-498c-be2e-b04d3b00cf12.png">

chraibi commented:
Windows do not like `Log.h`s definition of `ERROR`, since this constant is already defined somewhere else. 
Therefore, this suggestion [her](https://www.dropbox.com/s/bsdfx81pus1n4jt/Undef_Windows_defined_ERROR_.patch?dl=0).

chraibi reviewed (changes requested):
Great PR and thanks for your efforts. See comments!

chraibi reviewed (commented):
I think you might remove this [osx-fix](https://github.com/JuPedSim/jpsvis/tree/master/src/fix) as well.

Ozaq commented:
@chraibi
* Removed os_fix
* pushed a fix attempt for the error macro
* added compute file for vtk8/9

We will see how may more API changes are present in vtk.

chraibi commented on CMakeLists.txt:
```diff
@@ -88,6 +82,12 @@ add_executable(jpsvis MACOSX_BUNDLE WIN32 src/main.cpp ${app_icon_macos})
+target_compile_definitions(jpsvis
+    PRIVATE
+        $<$<CXX_COMPILER_ID:MSVC>:NOGDI>
+        $<$<CXX_COMPILER_ID:MSVC>:WIN32_LEAN_AND_MEAN>
```
It´s a kind of magic! :-)

chraibi commented on src/VtkVersionSwitch.h:
```diff
@@ -0,0 +1,19 @@
+/// This file provides wrapper functions around breakage in VTK API versions.
```
Sauber!

chraibi reviewed (approved):
builds succesfully on windows.

Ozaq commented:
@chraibi and @schroedtert I took the liberty to resolve all comments, I did address all your points.
